utils/litemap/src/map.rs

```diff
@@ -210,6 +210,21 @@ impl<K: Ord, V> IndexMut<&'_ K> for LiteMap<K, V> {
         self.get_mut(key).expect("LiteMap could not find key")
     }
 }
+impl<K: Ord, V> FromIterator<(K, V)> for LiteMap<K, V> {
+    fn from_iter<I: IntoIterator<Item = (K, V)>>(iter: I) -> Self {
+        let iter = iter.into_iter();
+        let mut map = match iter.size_hint() {
+            (lower, None) => LiteMap::with_capacity(lower),
+            (_, Some(upper)) => LiteMap::with_capacity(upper),
+        };
```

Comment:
`optional`: could you reverse the items? It makes my brain wrap when I reason about the behavior for some reason, and we really want to say here "If there is an upper limit, set it, if not, set the lower", so top-to-bottom order feels a bit easier for humans.

Reply:
Done :)